src/state/connection/index.ts

```diff
@@ -0,0 +1,62 @@
+import { signal } from '@preact-signals/safe-react';
+import { PenumbraRequestFailure, PenumbraState, PenumbraManifest } from '@penumbra-zone/client';
+import { penumbra, PRAX_ORIGIN } from '@/utils/penumbra';
+
+export const providerManifest = signal<PenumbraManifest>();
+
+export const providerConnected = signal<boolean>();
```

Comment:
refactored the `useConnect` to use signals – this way `providerConnected` can be used not only within react components but also in any external TS code